src/server/hll_family_test.cc

```diff
@@ -26,6 +29,20 @@ TEST_F(HllFamilyTest, Simple) {
   EXPECT_EQ(CheckedInt({"pfcount", "key"}), 1);
 }
 
+TEST_F(HllFamilyTest, Promote) {
+  int unique_values = 20000;
+  for (int i = 0; i < unique_values; ++i) {
+    std::string newkey = GenerateUniqueValue(i);
+    // sparse hll is promoted to dense at the 1660th+- insertion. But how to verify that?
+    Run({"pfadd", "key", newkey});
```

Comment:
It could be great to test adding multiple values at once (like the bug that I think I found). Can you randomize the number of entries to add, say between 1 and 20?
You could then do `i += count` instead of `++i` in the `for` loop to keep the size constant